Adds Prom and k8s configs to start scraping the script_exporter service.

nkinkade opened the pull request:
The PR adds the necessary configs for Prom to start scraping the new script_exporter service.

stephen-soltesz commented:
Reviewed 1 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[config/federation/prometheus/prometheus.yml.template, line 456 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-L2amBtwO9BuzaCrXJZT:-L2amBtwO9BuzaCrXJZU:bueujhx) ([raw file](https://github.com/m-lab/prometheus-support/blob/c1cf51e6164f2995eee7a530bdf153b94e98d0ba/config/federation/prometheus/prometheus.yml.template#L456)):*
> ```
>     file_sd_configs:
>       - files:
>           - /script-targets/*.json
> ```

Do we need to add this directory to the k8s config for prometheus?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/166)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[config/federation/prometheus/prometheus.yml.template, line 465 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-L2ammfDzJdZHJMdTD9Z:-L2ammfDzJdZHJMdTD9_:bdvxb4h) ([raw file](https://github.com/m-lab/prometheus-support/blob/c1cf51e6164f2995eee7a530bdf153b94e98d0ba/config/federation/prometheus/prometheus.yml.template#L465)):*
This link does not exist yet:

> https://github.com/m-lab/prometheus-script-exporter/blob/master/script_exporter.yml

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/166)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
@stephen-soltesz: I've addressed your comments, I believe. PTAL?

---

Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[config/federation/prometheus/prometheus.yml.template, line 456 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-L2amBtwO9BuzaCrXJZT:-L2aqKtw-aWK2_h1TL5G:b2hbtxu) ([raw file](https://github.com/m-lab/prometheus-support/blob/c1cf51e6164f2995eee7a530bdf153b94e98d0ba/config/federation/prometheus/prometheus.yml.template#L456)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

Do we need to add this directory to the k8s config for prometheus?
</blockquote></details>

A change to the k8s config was part of this PR. The file matrix at the top shows that you already reviewed that change?  You can [see the change in the Github PR](https://github.com/m-lab/prometheus-support/pull/166/files).

---

*[config/federation/prometheus/prometheus.yml.template, line 465 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-L2ammfDzJdZHJMdTD9Z:-L2apXdP3Ms0348pKp5C:bisi6p) ([raw file](https://github.com/m-lab/prometheus-support/blob/c1cf51e6164f2995eee7a530bdf153b94e98d0ba/config/federation/prometheus/prometheus.yml.template#L465)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

This link does not exist yet:

> https://github.com/m-lab/prometheus-script-exporter/blob/master/script_exporter.yml

</blockquote></details>

True. I was future-proofing this PR so that I didn't have to go back and change this later. [As soon as another PR gets merged](https://github.com/m-lab/prometheus-script-exporter/pull/1), this URL will be good. Is that okay?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-:-L2aqalcAlmJbK-iO0YW:b-r249nj)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[config/federation/prometheus/prometheus.yml.template, line 456 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-L2amBtwO9BuzaCrXJZT:-L2ar8ZDvr9XNY6LIpVL:b-scj4ye) ([raw file](https://github.com/m-lab/prometheus-support/blob/c1cf51e6164f2995eee7a530bdf153b94e98d0ba/config/federation/prometheus/prometheus.yml.template#L456)):*
<details><summary><i>Previously, nkinkade wrote…</i></summary><blockquote>

A change to the k8s config was part of this PR. The file matrix at the top shows that you already reviewed that change?  You can [see the change in the Github PR](https://github.com/m-lab/prometheus-support/pull/166/files).
</blockquote></details>

I'm talking about these bits: https://github.com/m-lab/prometheus-support/blob/master/k8s/prometheus-federation/deployments/prometheus.yml#L93 Do we need a new one for `/script-targets/`?

---

*[config/federation/prometheus/prometheus.yml.template, line 465 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-L2ammfDzJdZHJMdTD9Z:-L2arLbMa5Xs0BO47W85:by75rjg) ([raw file](https://github.com/m-lab/prometheus-support/blob/c1cf51e6164f2995eee7a530bdf153b94e98d0ba/config/federation/prometheus/prometheus.yml.template#L465)):*
<details><summary><i>Previously, nkinkade wrote…</i></summary><blockquote>

True. I was future-proofing this PR so that I didn't have to go back and change this later. [As soon as another PR gets merged](https://github.com/m-lab/prometheus-script-exporter/pull/1), this URL will be good. Is that okay?
</blockquote></details>

Yes, for sure. There was nothing on the radar 20 minutes ago :-)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/166)*
<!-- Sent from Reviewable.io -->

nkinkade commented:
PTAL?

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[config/federation/prometheus/prometheus.yml.template, line 456 at r1](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-L2amBtwO9BuzaCrXJZT:-L2ascMHQiVUcx_QiUHS:bty5ht1) ([raw file](https://github.com/m-lab/prometheus-support/blob/c1cf51e6164f2995eee7a530bdf153b94e98d0ba/config/federation/prometheus/prometheus.yml.template#L456)):*
<details><summary><i>Previously, stephen-soltesz (Stephen Soltesz) wrote…</i></summary><blockquote>

I'm talking about these bits: https://github.com/m-lab/prometheus-support/blob/master/k8s/prometheus-federation/deployments/prometheus.yml#L93 Do we need a new one for `/script-targets/`?
</blockquote></details>

Sorry. I misunderstood/forgot about that bit. I'm submitted another PR to fix that omission.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-:-L2asjfEmGFx3eW9LOK6:b-833305)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/166)*
<!-- Sent from Reviewable.io -->

stephen-soltesz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/m-lab/prometheus-support/166#-:-L2ayd3g0hr4tBdb4gMg:bnfp4nl)*
<!-- Sent from Reviewable.io -->
